src/include/test/unit_tdb_matrix.cc

```diff
@@ -271,3 +271,49 @@ TEST_CASE("tdb_matrix: MatrixBase template parameter", "[tdb_matrix]") {
     }
   }
 }
+
+TEST_CASE("tdb_matrix: empty matrix", "[tdb_matrix]") {
```

Comment:
We should add a test for oddball `start` and `stop` as well as for "half full" - `(start, std::nullopt`) or (`std::null opt, stop`) where the array has non-zero `start`.